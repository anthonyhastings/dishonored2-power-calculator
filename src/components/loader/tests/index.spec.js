import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../';

describe('Loader component', () => {
  let testContext;

  const renderComponent = (props = {}) => {
    return (
      <Loader {...props}>
        <p>Hello world.</p>
      </Loader>
    );
  };

  beforeEach(() => {
    testContext = {};

    testContext.defaultProps = {
      loadingState: true
    };
  });

  describe('when loading is incomplete', () => {
    beforeEach(() => {
      testContext.component = renderer.create(
        renderComponent({
          ...testContext.defaultProps,
          loadingState: false
        })
      );
    });

    it('renders the loader', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe('when loading is complete', () => {
    beforeEach(() => {
      testContext.component = renderer.create(
        renderComponent({
          ...testContext.defaultProps,
          loadingState: true
        })
      );
    });

    it('renders the loaders children', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe('when loading has errored', () => {
    beforeEach(() => {
      testContext.component = renderer.create(
        renderComponent({
          ...testContext.defaultProps,
          loadingState: new Error()
        })
      );
    });

    it('renders the loaders error children', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });
});
