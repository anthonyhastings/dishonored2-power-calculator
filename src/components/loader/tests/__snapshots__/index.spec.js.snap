// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component when loading has errored renders the loaders error children 1`] = `
<div
  className="app-loader"
>
  <h2
    className="app-loader__message"
  >
    Error while loading
  </h2>
  <div
    className="app-loader__icon-container"
  />
</div>
`;

exports[`Loader component when loading is complete renders the loaders children 1`] = `
<p>
  Hello world.
</p>
`;

exports[`Loader component when loading is incomplete renders the loader 1`] = `
<div
  className="app-loader app-loader--is-loading"
>
  <h2
    className="app-loader__message"
  >
    Loading
  </h2>
  <div
    className="app-loader__icon-container"
  />
</div>
`;
